File: src/actions/debug.js
"use strict";

const chalk = require("chalk");
const { exit, fmt } = require("../utils/toolbox.js");
const { initconfig, getsetting, setsetting } = require("../utils/config.js");

module.exports = async (args) => {
	let { enable, disable, script } = args;

	await initconfig();

	if (enable && disable) {
		let varg1 = chalk.bold("--enable");
		let varg2 = chalk.bold("--disable");
		let tstring = "? and ? given when only one can be provided.";
		exit([fmt(tstring, varg1, varg2)]);
	}

	// 0=off , 1=debug , 2=debug + ac.pl , 3=debug + ac.nim
	if (enable) {
		let dl = script === "nim" ? 3 : script === "pl" ? 2 : 1;
		await setsetting("debug", dl);
		console.log(chalk.green("on"));
	} else if (disable) {
		await setsetting("debug", 0);
		console.log(chalk.red("off"));
	} else {
		process.stdout.write(await getsetting("debug"));
	}
};

File: src/parser/nim/actions/debug.nim
import strutils, asyncdispatch, json

import ../utils/[chalk, config]

proc nlcli_debug*(s: string = "{}") {.async.} =
    let jdata = parseJSON(s)
    let enable = jdata{"enable"}.getStr()
    let disable = jdata{"disable"}.getStr()
    var script = jdata{"script"}.getStr()

    initconfig()

    if enable.len != 0 and disable.len != 0:
        let varg1 = "--enable".chalk("bold")
        let varg2 = "--disable".chalk("bold")
        let tstring = "$1 and $2 given when only one can be provided.";
        quit(tstring % [varg1, varg2])

    # 0=off , 1=debug , 2=debug + ac.pl , 3=debug + ac.nim
    if enable.len != 0:
        let dl = (
            if script == "nim": 3
            elif script == "perl": 2
            else: 1
        )
        setsetting("debug", $dl)
        echo "on".chalk("green")
    elif disable.len != 0:
        setsetting("debug", "0")
        echo "off".chalk("red")
    else:
        stdout.write(getsetting("debug"))
